### PR TITLE
Solves #16 - App Localization

### PR DESCRIPTION
**Pull request by aliyorulmazdev**
Will translate every component once u review this.

### CODE REVIEW

File: prExplorer/API/obj/Debug/net8.0/API.GeneratedMSBuildEditorConfig.editorconfig [hunk 1]
```diff
@@ -9,9 +9,10 @@ build_property.EnforceExtendedAnalyzerRules =
...
 build_property.RootNamespace = API
 build_property.RootNamespace = API
-build_property.ProjectDir = c:\Users\aliyo\source\repos\NC\prExplorer\API\
+build_property.ProjectDir = C:\Users\Ermat\projects\NC\prExplorer\API\
+build_property.EnableComHosting = 
 build_property.RazorLangVersion = 7.0
 build_property.SupportLocalizedComponentNames = 
 build_property.GenerateRazorMetadataSourceChecksumAttributes = 
-build_property.MSBuildProjectDirectory = c:\Users\aliyo\source\repos\NC\prExplorer\API
+build_property.MSBuildProjectDirectory = C:\Users\Ermat\projects\NC\prExplorer\API
```
Comment (dsibinski) [inline prExplorer/API/obj/Debug/net8.0/API.GeneratedMSBuildEditorConfig.editorconfig:17]: your local paths should not go in the public repo

File: prExplorer/client-app/src/app/layout/NavBar.tsx [hunk 2]
```diff
@@ -1,14 +1,19 @@
-import React from "react";
+import React, { useTransition } from "react";
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/layout/NavBar.tsx]: not needed I guess

File: prExplorer/client-app/public/locales/en/translation.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "products": "Products",
```
Comment (dsibinski) [inline prExplorer/client-app/public/locales/en/translation.json]: This is a good solution with i18n. However, consider that soon your translations might also be needed to use on the server. For example, for translating some exceptions or maybe you will send emails/notifications from your server and you want them to be translated as well.

In the future you can think of moving these translation files to the server. Then to use these common files both on the client and the server.

In .NET, you can use `IStringLocalizer`. More info here: https://learn.microsoft.com/en-us/dotnet/core/extensions/localization

**Review by dsibinski — APPROVED** [review]
@aliyorulmazdev small remarks, but generally it's good :)

File: prExplorer/client-app/public/locales/en/translation.json (hunk 3 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/public/locales/en/translation.json]: I can create translationcontroller and store these on database, serve as json. And use it on translationStore same as productStore. That would be good

File: prExplorer/API/obj/Debug/net8.0/API.GeneratedMSBuildEditorConfig.editorconfig (hunk 1 above)
Comment (aliyorulmazdev) [reply prExplorer/API/obj/Debug/net8.0/API.GeneratedMSBuildEditorConfig.editorconfig:17]: I added this file to gitignore but not works. Need to find another way

File: prExplorer/client-app/public/locales/en/translation.json (hunk 3 above)
Comment (dsibinski) [reply prExplorer/client-app/public/locales/en/translation.json]: Exactly, but also consider using `IStringLocalizer` for the server

**Pull request merged**